submissions/unabyband/OO-JS/app.js

```diff
@@ -0,0 +1,141 @@
+// Enemies our player must avoid
+// Variables applied to each of our instances go here,
+// we've provided one for you to get started
+// The image/sprite for our enemies, this uses
+// a helper we've provided to easily load images
+const Enemy = function(horizontalMove, verticalPosition, speed) {
+    this.horizontalMove = horizontalMove;
+    this.verticalPosition = verticalPosition;
+    this.speed = speed;
+    this.sprite = 'images/enemy-bug.png';
+};
+
+let winsCounter = 0;
+let speedIncreaser = 100;
+let mainMessage = "";
+let enemy = new Enemy();
+const allEnemies = []; 
+const enemyLocation = [63, 147, 230];
+const enemyStartPosition = -50;
+const enemyEndPosition = 510;
+const enemyWidth = 80;
+const enemyHeight = 60;
+const playerStartHorizontal = 202;
+const playerStartVertical = 405;
+const playerHorizontalStep = 102;
+const playerVerticalStep = 83;
+const maxRounds = 12;
+
+// Update the enemy's position, required method for game
+// Parameter: dt, a time delta between ticks
+// You should multiply any movement by the dt parameter
+// which will ensure the game runs at the same speed for
+// all computers.
+Enemy.prototype.update = function(dt) {
+    this.horizontalMove += this.speed * dt;
+    if (this.horizontalMove > enemyEndPosition) {
+        this.horizontalMove = enemyStartPosition;
+        this.speed = speedIncreaser + Math.floor(Math.random() * 222);
+    };
+    this.checkCollision(player);
```

Comment:
See item 4.iii [here](https://kottans.org/documentation/docs/doc/code-review/#oo-js-frogger)

Reply:
Done, thanks for your notice